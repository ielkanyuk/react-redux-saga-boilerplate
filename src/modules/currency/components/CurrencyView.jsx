import React from 'react';
import PropTypes from 'prop-types';

export class CurrencyView extends React.Component {
    static propTypes = {
        rate: PropTypes.number,
        pair: PropTypes.string,
        loading: PropTypes.bool.isRequired,
        update: PropTypes.func.isRequired,
        error: PropTypes.string,
    };

    handleUpdate = (e) => {
        const { update } = this.props;
        update({ currencyPair: e.target.value });
    }

    render() {
        const {
            rate, pair, loading, error,
        } = this.props;

        return (
            <div className="form">
                <div className="form_buttons">
                    <select onChange={this.handleUpdate} value={pair || ''}>
                        <option value="">---</option>
                        <option value="EUR_USD">EUR_USD</option>
                        <option value="EUR_RUB">EUR_RUB</option>
                        <option value="USD_RUB">USD_RUB</option>
                    </select>
                </div>
                <br />
                {error && <span>{error}</span>}
                {loading && <span>Loading...</span>}
                {rate && (
                    <div className="form_group">
                        <div>
                            PAIR: {pair}, VALUE: {rate}
                        </div>
                    </div>
                )}
            </div>
        );
    }
}