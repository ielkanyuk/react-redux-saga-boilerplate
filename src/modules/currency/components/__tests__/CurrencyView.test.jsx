/* eslint-env jest/globals */
import React from 'react';
import renderer from 'react-test-renderer';
import { CurrencyView } from '../CurrencyView';

it('renders correctly', () => {
    const tree = renderer.create(<CurrencyView
        loading={false}
        update={() => {
        }}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});