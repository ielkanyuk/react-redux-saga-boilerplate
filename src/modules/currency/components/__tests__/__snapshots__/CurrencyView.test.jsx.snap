// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form"
>
  <div
    className="form_buttons"
  >
    <select
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        ---
      </option>
      <option
        value="EUR_USD"
      >
        EUR_USD
      </option>
      <option
        value="EUR_RUB"
      >
        EUR_RUB
      </option>
      <option
        value="USD_RUB"
      >
        USD_RUB
      </option>
    </select>
  </div>
  <br />
</div>
`;
